Classes/Backend/TceMain.php

```diff
@@ -80,6 +80,91 @@ public function processDatamap_postProcessFieldArray($status, $table, $id, &$fie
+        $doProcessCommand = $doProcessCommand && 'tt_content' == $table;
+
+
+        if ($doProcessCommand) {
+            // A copy, or localisation (which is also a copy) was made. Cascade copy operations for child records.
+            $gridChildRecords = [];
+            $resolver = $this->objectManager->get(ProviderResolver::class);
+            $originalRecord = $this->recordService->getSingle($table, '*', $id);
+            $recordCanBeGrid = !empty($originalRecord['pi_flexform']);
+            if(!$recordCanBeGrid){
+                return;
+            }
+
+
+            $primaryProvider = $resolver->resolvePrimaryConfigurationProvider(
+                $table,
+                null,
+                $originalRecord
+            );
+            $colPosCalcBaseUid = $id;
+            $gridIsTranslation = $originalRecord['sys_language_uid'] > 0;
+            if($gridIsTranslation){
+                $colPosCalcBaseUid = $originalRecord['l18n_parent'];
+            }
+            if ($primaryProvider) {
+                $childColPosValues = [];
+                foreach ($primaryProvider->getGrid($originalRecord)->getRows() as $row) {
+                    foreach ($row->getColumns() as $column) {
+                        $childColPosValues[] = ColumnNumberUtility::calculateColumnNumberForParentAndColumn(
+                            $colPosCalcBaseUid,
+                            $column->getColumnPosition()
+                        );
+                    }
+                }
+
+                // Selecting records to copy. The "sorting DESC" is very intentional, since we are copying children
+                // into columns by consistently placing them in the topmost position. When copying is complete,
+                // children will have the exact opposite order of the "sorting DESC" result - which means they are
+                // sorted correctly, ascending, as the original child records were.
+                if (!empty($childColPosValues)) {
+                    $gridChildRecords = $this->recordService->get(
+                        $table,
+                        'uid, colPos, pid, sorting',
+                        sprintf(
+                            'colPos IN (%s) and sys_language_uid=%s',
+                            implode(', ', $childColPosValues),
+                            $originalRecord['sys_language_uid']
+                        ),
+                        null,
+                        'sorting DESC'
+                    );
+                }
+            }
+            $gridHasChildren = count($gridChildRecords) > 0;
+
+            if('delete' == $command && $gridHasChildren){
+                $cmd = [
+                    'tt_content' => []
+                ];
+                foreach ($gridChildRecords as $recordToDelete) {
+                    $cmd['tt_content'][ $recordToDelete['uid'] ] = ['delete' => 1];
+                }
+                $dataHandler = GeneralUtility::makeInstance(DataHandler::class);
```

Comment:
Call on $reference instead - and I think this needs to be different. This is the command pre-processing method so we can simply extend the command map array on $reference and it should then do those, between calling this method and the post-processing method.